atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraRequestExceptionHandlerTest.java

```diff
@@ -29,100 +31,168 @@
 import org.apache.cassandra.thrift.TimedOutException;
 import org.apache.cassandra.thrift.UnavailableException;
 import org.apache.thrift.transport.TTransportException;
+import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
 
 import com.google.common.collect.Iterables;
 import com.google.common.collect.Sets;
 import com.palantir.atlasdb.cassandra.CassandraKeyValueServiceConfig;
 import com.palantir.atlasdb.keyvalue.api.InsufficientConsistencyException;
 
+@RunWith(Parameterized.class)
 public class CassandraRequestExceptionHandlerTest {
-
     private static final String MESSAGE = "a exception";
     private static final Exception CAUSE = new Exception();
     private static final CassandraKeyValueServiceConfig config = mock(CassandraKeyValueServiceConfig.class);
     private static final int MAX_RETRIES_PER_HOST = 3;
     private static final int MAX_RETRIES_TOTAL = 6;
 
-    private Set<Exception> connectionExceptions = Sets.newHashSet(new SocketTimeoutException(MESSAGE),
+    private static Set<Exception> connectionExceptions = Sets.newHashSet(new SocketTimeoutException(MESSAGE),
             new CassandraClientFactory.ClientCreationFailedException(MESSAGE, CAUSE));
-    private Set<Exception> transientExceptions = Sets.newHashSet(new TTransportException(), new TimedOutException(),
-            new InsufficientConsistencyException(MESSAGE));
-    private Set<Exception> indicativeOfCassandraLoadException = Sets.newHashSet(new NoSuchElementException(),
-            new UnavailableException());
-    private Set<Exception> fastFailoverExceptions = Sets.newHashSet(new InvalidRequestException());
+    private static Set<Exception> transientExceptions = Sets.newHashSet(new TTransportException());
+    private static Set<Exception> indicativeOfCassandraLoadException = Sets.newHashSet(new NoSuchElementException(),
+            new TimedOutException(), new UnavailableException(), new InsufficientConsistencyException(MESSAGE));
+    private static Set<Exception> fastFailoverExceptions = Sets.newHashSet(new InvalidRequestException());
+
+    private boolean currentMode;
+    private CassandraRequestExceptionHandler handler;
 
-    private CassandraRequestExceptionHandler exceptionHandler = new CassandraRequestExceptionHandler(
-            () -> MAX_RETRIES_PER_HOST,
-            () -> MAX_RETRIES_TOTAL,
-            new Blacklist(config));
+    @Parameterized.Parameters
+    public static Iterable<? extends Object> data() {
+        return Arrays.asList(false, true);
+    }
+
+    @Parameterized.Parameter
```

Comment:
I think this test would be easier to read and maintain if we just copied all the tests for the different handlers.  And when we delete one later it is easier to remove too.

Reply:
done